test/python/transpiler/test_permrowcol.py

```diff
@@ -0,0 +1,133 @@
+
+    def test_choose_row_returns_int(self):
+        """Test the output type of choose_row"""
+        coupling = CouplingMap()
+        permrowcol = PermRowCol(coupling)
+        parity_mat = np.array([[1, 0], [0, 1]])
+        vertices = np.array([0, 1])
+
+        instance = permrowcol.choose_row(vertices, parity_mat)
+
+        self.assertIsInstance(instance, np.int64)
+
+    def test_choose_row_returns_correct_index(self):
+        """
+        Test method to test the correctness of the choose_row method
+        """
+        coupling = CouplingMap()
+        permrowcol = PermRowCol(coupling)
+        parity_mat = np.array(
+            [
+                [0, 1, 0, 0, 0, 1, 0, 0, 0],
+                [1, 0, 1, 0, 1, 0, 0, 0, 0],
+                [0, 1, 0, 0, 0, 0, 0, 0, 0],
+                [1, 0, 1, 0, 1, 0, 1, 0, 1],
+                [0, 1, 0, 1, 0, 1, 0, 1, 0],
+                [1, 0, 0, 0, 1, 0, 1, 0, 0],
+                [0, 0, 0, 0, 0, 1, 0, 1, 0],
+                [0, 0, 1, 0, 0, 0, 1, 0, 1],
+                [0, 1, 0, 1, 0, 1, 0, 1, 0],
+            ]
+        )
+        vertices = np.array([1, 3, 4, 5, 6, 8])
+
+        index = permrowcol.choose_row(vertices, parity_mat)
+
+        self.assertEqual(index, 6)
+
+    def test_choose_column_returns_int(self):
+        """Test the output type of choose_column"""
+        coupling = CouplingMap()
+        permrowcol = PermRowCol(coupling)
+        parity_mat = np.array([[1, 0], [0, 1]])
+        cols = np.array([0, 1])
+
+        instance = permrowcol.choose_column(parity_mat, cols, 0)
+
+        self.assertIsInstance(instance, np.int64)
+
+    def test_choose_column_returns_correct_index(self):
+        """Test choose_colum method for correctness"""
+        coupling = CouplingMap()
+        permrowcol = PermRowCol(coupling)
+
+        parity_mat = np.array(
+            [
+                [0, 1, 0, 0, 0, 1, 0, 0, 0],
+                [0, 0, 0, 0, 1, 0, 0, 0, 0],
+                [0, 1, 0, 0, 0, 0, 0, 0, 0],
+                [1, 0, 1, 0, 1, 0, 1, 0, 1],
+                [0, 1, 0, 1, 0, 1, 0, 1, 0],
+                [1, 0, 0, 0, 1, 0, 1, 0, 0],
+                [0, 0, 0, 0, 0, 1, 0, 1, 0],
+                [0, 0, 1, 0, 0, 0, 1, 0, 1],
+                [0, 1, 0, 1, 0, 1, 0, 1, 0],
+            ]
+        )
+        vertices = np.array([1, 3, 4, 5, 6, 7])
+
+        index = permrowcol.choose_column(parity_mat, vertices, 4)
+
+        self.assertEqual(index, 3)
+
+    def test_choose_column_returns_correct_index_with_similar_col_sums(self):
+        """Test choose_column method for correctness in case of col_sums having same integers"""
+        coupling = CouplingMap()
+        permrowcol = PermRowCol(coupling)
+
+        parity_mat = np.array([[1, 0, 1], [0, 1, 1], [0, 0, 1]])
+
+        vertices = np.array([0, 1, 2])
+
+        index = permrowcol.choose_column(parity_mat, vertices, 2)
+
+        self.assertEqual(index, 2)
+
+    def test_eliminate_column_returns_int(self):
```

Comment:
The name is out of date since it actually tests if it is an instance of np.ndarray